drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/force_valid_enrollment_coc.rb

```diff
@@ -0,0 +1,56 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+module HmisCsvTwentyTwenty::HmisCsvCleanup
+  class ForceValidEnrollmentCoc < Base
+    def cleanup!
+      enrollment_coc_batch = []
+
+      enrollment_coc_scope.find_each do |e_coc|
+        # ignore any valid CoC-codes
+        next if ::HUD.valid_coc?(e_coc.CoCCode)
+
+        # upcase any that match the format but aren't correctly cased
```

Comment:
I think that instead of if/elseif/else, we should do this as

1. add the dash if it is 2 letters and 3 digits
2. upcase the letters if it correctly formatted, but not cased
3. clear it if it isn't a valid CoC

I think the current logic will turn 'ma500' into 'ma-500' and then throw it out?

Reply:
adjusted